java-frontend/src/main/java/org/sonar/java/se/checks/LocksNotUnlockedCheck.java

```diff
@@ -173,10 +173,25 @@ public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager co
     ExplodedGraph.Node node = context.getNode();
     context.getState().getValuesWithConstraints(Status.LOCKED).keySet().stream()
       .flatMap(sv -> FlowComputation.flow(node, sv, ObjectConstraint.statusPredicate(Status.LOCKED), ObjectConstraint.statusPredicate(Status.UNLOCKED)).stream())
-      .forEach(loc -> reportIssue(reportOn(loc.syntaxNode), "Unlock this lock along all executions paths of this method.", Collections.emptySet()));
+      .forEach(this::reportIssue);
   }
 
-  private static Tree reportOn(Tree syntaxNode) {
+  private void reportIssue(JavaFileScannerContext.Location location) {
+    String flowMsg = "Lock " + quoteOrEmpty(lockName(location.syntaxNode)) + "is never unlocked";
+    Tree tree = issueTree(location.syntaxNode);
+    reportIssue(tree, "Unlock this lock along all executions paths of this method.", FlowComputation.singleton(flowMsg, tree));
+  }
+
+  private static String lockName(Tree syntaxNode) {
+    String name = SyntaxTreeNameFinder.getName(syntaxNode, SyntaxTreeNameFinder.Option.MEMBER_SELECT_CHOOSE_EXPR);
```

Comment:
This whole option handling is completely over-engineering for a pretty simple problem and in my opinion useless. According to the check (L#110), you already know that only method invocations on variable (form `var.lock()`) will be detected. So you don't have to use a lot of things to get the variable name from there...